### DISCUSSION TITLE
Exporting / importing whole configuration?

### DISCUSSION THREAD

**Discussion by ejgh-oe** [Security-Onion-Solutions/securityonion#13512] [Unsupported Versions]
Hi,
I'd like to set up the manager node of an existing installation on a fresh hardware (basically install from scratch). 

Is it possible to export the whole configuration from one (old) manager node and importing it in one piece to the new manager node, i.e. everything from suricata rules, variables, user accounts, SOC config etc.? 

In the docs I found "Daily backups create a tar file located in the `/nsm/backup/` directory located on the manager. "
(https://docs.securityonion.net/en/2.4/backup.html), but can this tar file be used for importing a configuration to a fresh installation?

Thanks much for any clue...

**Comment by cm-ops** [comment]
Are you just looking for the customizations made? If that is the case, https://docs.securityonion.net/en/2.4/backup.html will have all your cusomizations from `/opt/so/saltstack/local/` backed up. You can use these to place customizations back into the componenets soc_*.sls files.

The Kratos db and elasticsearch users in `/opt/so/saltstack/local/salt/elasticsearch/files/users` are backed up, but a fresh install will set different passwords for so_elastic, so_kibana, etc. The old users won't be synchronized with Elastic.

**Comment by ejgh-oe** [comment]
> Are you just looking for the customizations made? If that is the case, https://docs.securityonion.net/en/2.4/backup.html will have all your cusomizations from /opt/so/saltstack/local/ backed up. You can use these to place customizations back into the componenets soc_*.sls files.

I've made modifications mainly to

- pcap filters ( Administration > Configuration > bpf > pcap)
- BPF-filters for suricata (Administration > Configuration > bpf > suricata)
- IDH-exceptions (Config > advanced > idh > advanced)
- Custom suricata rules (file containing the rules is in /nsm/rules/detect-suricata/custom_file with SOC > config > server > modules > suricataengine > costomRulesets pointing to the file)

So are these settings being backed up at all?

Idea: Maybe it would be possible for a future release to have something like "config backup" in the GUI under Administration - would definitely help a lot. Not only for moving an installation to a new machine but simply for backup purposes.

**Reply by cm-ops** [reply]
Those modifications will be in the local pillars so they will get backed up. Below is what gets backed up in `/nsm`

```
nsm/kratos/
nsm/kratos/db/
nsm/kratos/db/db.sqlite
```
`/nsm/rules/detect-suricata/custom_file` would need to be backed up manually to replace custom rules.

**Reply by ejgh-oe** [reply]
Ah, thanks - I'll give it a shot in a "lab setup", i.e. setting up an isolated manager node and then trying to import the files/directories you mentioned.

**Comment by jstore-embers** [comment]
I would be interested in this as well. Mainly looking for a method to export the customizations made to the detections (suricata rule modifications and custom sigma filters).
